fvm/transactionEnv.go

```diff
@@ -115,53 +115,96 @@ func NewTransactionEnvironment(
 		env.seedRNG(ctx.BlockHeader)
 	}
 
-	err := env.setMeteringWeights()
+	// set the execution parameters from the state
+	err := env.setExecutionParameters()
 
 	return env, err
 }
 
-func (e *TransactionEnv) setMeteringWeights() error {
+func (e *TransactionEnv) setExecutionParameters() error {
+	// Check that the service account exists because all the settings are stored in it
+	serviceAddress := e.Context().Chain.ServiceAddress()
+	service := runtime.Address(serviceAddress)
+	ok, err := e.accounts.Exists(serviceAddress)
+	if err != nil {
+		_, fatal := errors.SplitErrorTypes(err)
+		if fatal != nil {
+			e.ctx.Logger.
+				Error().
+				Err(fatal).
+				Msgf("error reading execution parameters")
+			return fatal
+		}
+		if err != nil {
+			e.ctx.Logger.
+				Info().
+				Err(err).
+				Msgf("could not get service account to read execution parameters")
+			return nil
+		}
+	}
+	if !ok {
+		return nil
+	}
```

Comment:
not sure what this condition does

Reply:
The service account should exist, but there are cases when it doesn't. Besides tests, the one that comes to mind is bootstrapping. I think the first step during bootstrapping is to create the service account, I don't know how that work if this logic is hit before it exists.